### ISSUE TITLE
`io.quarkus:quarkus-bom:3.5.1` contains unresolved versions: `org.jboss:jdk-misc:${version.jdk-misc}`

### ISSUE THREAD

**Issue by ppalaga** [quarkusio/quarkus#36952]
### Describe the bug

We use a custom BOM flattener in Quarkus CXF and Camel Quarkus that is rather sensitive to broken source BOMs. 
After the upgrade to Quarkus 3.5.1, it started failing due to `version.jdk-misc` property being used but not being defined:

```
[ERROR] Failed to execute goal org.l2x6.cq:cq-maven-plugin:4.4.5:flatten-bom (flatten-bom) on project quarkus-cxf-bom: Execution flatten-bom of goal org.l2x6.cq:cq-maven-plugin:4.4.5:flatten-bom failed: Additional BOM io.quarkus:quarkus-bom:3.5.1 contains unresolved versions:
[ERROR]     - org.jboss:jdk-misc:${version.jdk-misc}
```

This is caused by https://github.com/jboss/jboss-parent-pom/issues/236 and the fix is underway via https://github.com/jboss/jboss-parent-pom/pull/237

I wonder if there is any way to workaround this on the Quarkus side while waiting for jboss-parent 41? 
I mean `flatten-platform-bom` could perhaps be configured to ignore `org.jboss:jdk-misc` via `excludeArtifactKeys`. WDYT @aloubyansky ?

**Comment by ppalaga** [comment]
> `flatten-platform-bom` could perhaps be configured to ignore `org.jboss:jdk-misc` via `excludeArtifactKeys`. WDYT @aloubyansky ?

It seems to work. Let me send a PR.

**Comment by aloubyansky** [comment]
`flatten-platform-bom` also has `filterInvalidConstraints` option, which we don't enable by default to save time on builds.

**Comment by aloubyansky** [comment]
I guess a better fix would be to update to the fixed jboss-parent

**Comment by ppalaga** [comment]
> I guess a better fix would be to update to the fixed jboss-parent

Well, if it is not released before Quarkus 3.6.0, then we'll have to implement a workaround in our flattener.

This change does exactly what we need and I do not see any time penalty: https://github.com/quarkusio/quarkus/pull/36953/files 

Another possible workaround would be to add `<version.jdk-misc>3.Final</version.jdk-misc>` to quarkus BOM.

**Comment by gsmet** [comment]
We have merged a workaround for now.
